src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc

```diff
@@ -443,8 +443,8 @@ XdsResolver::XdsConfigSelector::XdsConfigSelector(
   // weighted_cluster_state field points to the memory in the route field, so
   // moving the entry in a reallocation will cause the string_view to point to
   // invalid data.
-  route_table_.reserve(resolver_->current_virtual_host_.routes.size());
-  for (auto& route : resolver_->current_virtual_host_.routes) {
+  route_table_.reserve(resolver_->current_virtual_host_->routes.size());
```

Comment:
nit: I am slightly uncomfortable with the fact that current_virtual_host_ that may be unset to be unreferenced unconditionally.

Reply:
This is guaranteed never to be called when `current_virtual_host_` is unset, because the only place we call it is from `GenerateResult()`, which explicitly checks whether `current_virtual_host_` is set first.

Reply:
Yes, I understand - my usual concern is that it might bite later, after the code gets reshuffled for some reason...

Reply:
Yeah, I agree.  In general, I prefer to handle this kind of thing by passing the value in by parameter, but the existing code here already has the `XdsConfigSelector` ctor reaching into a lot of the resolver's internal state, and some of those operations are writes, not just reads.  So in this particular case, there's already a bunch of shared logic between the two objects, so there's not much point in trying to pretend otherwise.